conftest.py

```diff
@@ -176,37 +168,61 @@ def function_to_seed_torch(seed):
 
 
 @pytest.fixture(autouse=True)
-def autoseeding_of_everything(record_property, request):
+def autoseeding_of_everything(request):
     """Seed everything we can, for determinism."""
-    main_seed = request.config.getoption("--forcing_random_seed", default=None)
+    randomly_seed = request.config.getoption("--randomly-seed", default=None)
 
-    if main_seed is None:
-        main_seed = random.randint(0, 2**64 - 1)
+    if randomly_seed is None:
+        raise ValueError("why isn't --randomly-seed set?")
 
-    seed = main_seed
-    record_property("main seed", main_seed)
+    # We need to find the relative file path of the test file. It does not look native with request,
+    # so we recompute it.
+    absolute_path = str(request.fspath)
+
+    assert (
+        absolute_path.count("concrete-ml/test") == 1
+    ), f"the absolute path is not as expected {absolute_path=}"
+
+    relative_file_path = absolute_path[
+        absolute_path.find("concrete-ml/test") + len("concrete-ml/") :
+    ]
+
+    # Derive the sub_seed from the randomly_seed and the test name
+    derivation_string = f"{relative_file_path} # {str(request.node.name)} # {randomly_seed}"
+
+    hash_object = hashlib.sha256()
+    hash_object.update(b"{derivation_string}")
+    hash_object.digest()
+    hash_value = hash_object.hexdigest()
+
+    sub_seed = int(hash_value, 16) % 2**64
```

Comment:
why is this needed ?

Reply:
why _what_ is needed? which part?

Reply:
well, the line I comment on : `    sub_seed = int(hash_value, 16) % 2**64`

is it because the has value (in hexa) can go above 2**64 b ? I would jut add a comment around this part (and the above lines) to add a simple `# Get the subseed by getting the hash value and retrieving an int64 value`

also : 
- for seeding I thought numpy prefered 32b values (in the code we often use % 2**31 for seed iirc)
- I guess the risk of collisiions because of this %64 are minimal, right  ?

Reply:
- the hash is a SHA256, so it's 256b. And random.seed wants a 64b seed and numpy.random.seed wants a 32 seed; so I reduce to 64b and I have `numpy.random.seed(seed % 2**32)`
- using % 2**31 reduces to 31b, there is one bit loss here
- we have collisions on 64b in O(2**32) but it's not a problem here [what's the attack] and it's not because of us [it's python and numpy which want reduced seeds]

Reply:
yeah got it thanks !  well I would just add a little line to simply explain that, nothing fancy

Reply:
Adding a line.

Reply:
great !